### PR TITLE
Builder improvements

### PR DESCRIPTION
**Pull request by michamos**
Closes inspirehep/hepcrawl#176.

### CODE REVIEW

File: inspire_schemas/builders/literature.py [hunk 1]
```diff
@@ -48,7 +48,8 @@ def func_wrapper(self, *args, **kwargs):
                      if value not in EMPTIES}
 
         if (
-            list(my_kwargs.keys()) == ['source'] or not list(my_kwargs.keys())
+            set(my_kwargs.keys()).issubset(['source', 'material']) or
+                not list(my_kwargs.keys())
```
Comment (kaplun) [inline inspire_schemas/builders/literature.py]: :boom:

Comment (kaplun) [reply inspire_schemas/builders/literature.py]: You can slightly simplify with `(set(['source', 'material']) - set(my_kwargs)) or not my_kwargs`.

File: inspire_schemas/builders/literature.py [hunk 2]
```diff
@@ -194,7 +195,8 @@ def add_doi(self, doi, source=None, material=None):
         :param material: material for the doi.
         :type material: string
         """
-        if not idutils.normalize_doi(doi):
+        doi = idutils.normalize_doi(doi)
+        if not doi:
```
Comment (kaplun) [inline inspire_schemas/builders/literature.py]: This seems to fix a bug that is not documented in the commit log though.

Comment (michamos) [reply inspire_schemas/builders/literature.py]: :+1: forgot about that one. Will also add a test.

File: inspire_schemas/builders/literature.py (hunk 1 above)
Comment (michamos) [reply inspire_schemas/builders/literature.py]: actually that doesn't work. I want it to be true when `my_kwargs` is a subset of `{'source', 'material'}`, which is not the case for your proposal (or its negation).

Comment (kaplun) [reply inspire_schemas/builders/literature.py]: That's why the :boom: :smile:

**Pull request merged**